omnibus/config/software/openssl3.rb

```diff
@@ -25,17 +25,22 @@
 
 default_version "3.5.4"
 
+source url: "https://www.openssl.org/source/openssl-#{version}.tar.gz", extract: :lax_tar
+
+version("3.5.4") { source sha256: "967311f84955316969bdb1d8d4b983718ef42338639c621ec4c34fddef355e99" }
+
+relative_path "openssl-#{version}"
+
 build do
-  if (windows? && ENV["AGENT_FLAVOR"] != "fips") || (!windows? && ENV["AGENT_FLAVOR"] != "fips")
-    installation_dir = "#{install_dir}/embedded"
-    if windows?
-      installation_dir = "#{python_3_embedded}"
-    end
+  if !fips_mode?
+    installation_dir = if windows? then python_3_embedded else "#{install_dir}/embedded" end
     command_on_repo_root "bazelisk run -- @openssl//:install --destdir=#{installation_dir}"
     if windows?
-      # shutil generates temporary files during run install that are not removed afterwards.
-      Dir.glob("#{installation_dir}/include/openssl/tmp*").each do |tmp_file|
-        delete tmp_file
+      block do
+        # shutil generates temporary files during run install that are not removed afterwards.
+        Dir.glob("#{installation_dir}/include/openssl/tmp*").each do |tmp_file|
+          delete tmp_file
```

Comment:
Strictly speaking, since you're inside a `block`, you should avoid using `DSL` commands such as `delete` (which will "schedule" the deletion rather than do it immediately), here you probably want `FileUtils.rm`.

It's true that it won't matter in practice here because this is the last thing we do anyway.

In terms of the actual files that are left around, are we going to try to address this upstream?